File: 12915.java
import java.util.*;
class Solution {
  public String[] solution(String[] strings, int n) {
      String[] answer = {};
      
      for(int i = 0 ; i < strings.length ; i++){
          Character temp = strings[i].charAt(n);
          strings[i] = temp+strings[i];
      }
      Arrays.sort(strings);
      for (int i = 0 ; i < strings.length ; i++){
          strings[i] = strings[i].substring(1,strings[i].length());
          // System.out.println(strings[i]);
      }
      
      
      
      
      
      return strings;
  }
}

File: 12915.swift
func solution(_ strings:[String], _ n:Int) -> [String] {
    var strArray = strings;
    for i in 0..<strings.count{
        let index = strings[i].index(strings[i].startIndex, offsetBy: n)
        strArray[i] = "\(strings[i][index])"+strings[i]
    }
    strArray.sort()
    for i in 0..<strArray.count{
        strArray[i] = String(strArray[i].dropFirst())
    }
    return strArray
}
